core/src/mender-client.c

```diff
@@ -1237,10 +1220,7 @@ mender_client_update_work_function(void) {
                 mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_REBOOTING);
 
                 /* Save deployment data to publish deployment status after rebooting */
-                if (NULL == (storage_deployment_data = cJSON_PrintUnformatted(mender_client_deployment_data))) {
-                    mender_log_error("Unable to serialize deployment data for saving");
-                    ret = MENDER_FAIL;
-                } else if (MENDER_OK != (ret = mender_storage_set_deployment_data(storage_deployment_data))) {
+                if (MENDER_OK != mender_storage_set_deployment_data(mender_client_deployment_data)) {
```

Comment:
```suggestion
                if (MENDER_OK != (ret  = mender_storage_set_deployment_data(mender_client_deployment_data))) {
```
I think we need to assign the return value to `ret` here.